src/components/application_manager/test/commands/mobile/get_system_capability_request_test.cc

```diff
@@ -0,0 +1,246 @@
+namespace am = ::application_manager;
+namespace amt = application_manager_test;
+using am::commands::GetSystemCapabilityRequest;
+using am::commands::MessageSharedPtr;
+using am::event_engine::Event;
+namespace mobile_result = mobile_apis::Result;
+
+namespace {
+const uint32_t kConnectionKey = 2u;
+}  // namespace
+
+class GetSystemCapabilityRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  MessageSharedPtr CreateMessageParams() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    (*msg)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+    return msg;
+  }
+};
+
+TEST_F(GetSystemCapabilityRequestTest, Run_NotExistApp_AppNotRegistered) {
+  MockAppPtr mock_app;
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  MessageSharedPtr msg = CreateMessageParams();
+
+  SharedPtr<GetSystemCapabilityRequest> command(
+      CreateCommand<GetSystemCapabilityRequest>(msg));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(
+          MobileResultCodeIs(mobile_apis::Result::APPLICATION_NOT_REGISTERED),
+          _)).WillOnce(Return(true));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(GetSystemCapabilityRequestTest, Run_EmptyMessage_ExpectInvalidData) {
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  MessageSharedPtr msg = CreateMessageParams();
+  SharedPtr<GetSystemCapabilityRequest> command(
+      CreateCommand<GetSystemCapabilityRequest>(msg));
+
+  EXPECT_CALL(app_mngr_,
+              ManageMobileCommand(
+                  MobileResultCodeIs(mobile_apis::Result::INVALID_DATA), _))
+      .WillOnce(Return(true));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(GetSystemCapabilityRequestTest, Run_NAVIGATION_SUCCESS) {
+  MockAppPtr mock_app = CreateMockApp();
+  SmartObject sm(smart_objects::SmartType_Map);
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  MessageSharedPtr msg = CreateMessageParams();
+  amt::MockHMICapabilities mock_hmi_cap;
+  (*msg)[am::strings::msg_params][am::strings::system_capability_type] =
+      mobile_apis::SystemCapabilityType::NAVIGATION;
+  SharedPtr<GetSystemCapabilityRequest> command(
+      CreateCommand<GetSystemCapabilityRequest>(msg));
+  EXPECT_CALL(app_mngr_, hmi_capabilities()).WillOnce(ReturnRef(mock_hmi_cap));
+  EXPECT_CALL(mock_hmi_cap, navigation_capability())
+      .WillRepeatedly(Return(&sm));
+  EXPECT_CALL(app_mngr_,
+              ManageMobileCommand(MobileResultCodeIs(mobile_result::SUCCESS),
+                                  _)).WillOnce(Return(true));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(GetSystemCapabilityRequestTest, Run_SystemCapabilityPhoneCall_SUCCESS) {
+  MockAppPtr mock_app = CreateMockApp();
+  SmartObject sm(smart_objects::SmartType_Map);
```

Comment:
@ValeriiMalkov rename sm. This name is not obvious

Reply:
@LuxoftAKutsan fixed at the commit ab40e21d3a66941f9a9d9d0a0eac8e18f09add44